### PR TITLE
Enhance coding style with pep8 and fix `np.complex` depreciated function.

### PR DESCRIPTION
**Pull request by nennigb**
- Lint most python files (not to all to limit conflict with forth comming PR) with pep8
  - Fix `np.complex` depreciated function
  - Update CI script for ubuntu 22.04. build petsc with pip is still fragile and seems to fails with mpich. 
  - Skip doctest involving subprocess and mpi since it was not robust especially in conda env.

Merging this PR will ensure that all tests will pass on the CI.

### CODE REVIEW

File: eastereig/adapter.py [hunk 1]
```diff
@@ -15,20 +14,20 @@
 
 # You should have received a copy of the GNU General Public License
 # along with Eastereig.  If not, see <https://www.gnu.org/licenses/>.
+"""
+##Define an adapter design patern using object composition.
 
+It makes easier the implementation of linear algebra operation without concrete
+reference to specific library.
 
+Resused from :
```
Comment (MartinGhienne) [inline eastereig/adapter.py:23]: reused

File: eastereig/adapter.py [hunk 2]
```diff
@@ -37,118 +36,115 @@ class Adapter(object):
...
     # create the adapted objet
-    return Adapter(obj,**ADAPTER)
-    
+    return Adapter(obj, **ADAPTER)
+
+
 def adaptMat(obj, lib):
-    """ Create an Matrix-adapter for the different linear algera library
-    
+    """Create an Matrix-adapter for the different linear algera library.
```
Comment (MartinGhienne) [inline eastereig/adapter.py:110]: a matrix-adapter
libraries

**Review by MartinGhienne — APPROVED** [review]
Good job ! Just some typos to be fixed

**Comment by nennigb** [comment]
Big spring cleaning ;-)

**Pull request merged**